ddnoise.js

```diff
@@ -53,17 +53,22 @@ define(['./utils', 'underscore', 'promise'], function (utils, _) {
     };
 
     DdNoise.prototype.oneShot = function (sound) {
-        var source = this.context.createBufferSource();
+        var duration = sound.duration;
+        var context = this.context;
+        if (context.state !== "running") return duration;
+        var source = context.createBufferSource();
         source.buffer = sound;
         source.connect(this.gain);
         source.start();
-        return sound.duration;
+        return duration;
     };
 
     DdNoise.prototype.play = function (sound, loop) {
         var self = this;
+        var context = self.context;
+        if (context.state !== "running") return null;
         return new Promise(function (resolve, reject) {
-            var source = self.context.createBufferSource();
+            var source = context.createBufferSource();
```

Comment:
Where is `context` coming from now? I can't see that there's a closure-captured `context` anywhere - does this work? :) I probably misunderstand Javascript binding rules (As usual...)

One day I'll ES6-i-fy all this... I've only been using ES6 recently and all this `var self` and `prototype.blah` is so painful to read :)

Reply:
I tweaked it a little for clarity. It was working due to the inner function having access to the outer function's local variables. Same trick as the "var self = this" idiom but not as clear.